src/main/daml/Daml/Finance/Interface/Instrument/Option/Types.daml

```diff
@@ -10,3 +10,15 @@ data OptionTypeEnum
   | Put
     -- ^ Put option.
   deriving (Eq, Show)
+
+-- | A barrier type classification.
+data BarrierTypeEnum
+  = UpAndOut
```

Comment:
At the moment we only have the `Lte` inequality operator. I wonder if this is enough to cover all these cases (or if we need to introduce `Gte` as well). What do you think?

Reply:
Is it this part of the code that you are concerned does not cover all cases?

```
    barrierEvent = if isDown then Lte (Observe spot, Const barrier)
                             else Lte (Const barrier, Observe spot)
```

I think it should cover terms of the type "if the underlying ever trades at or below (above) the barrier, then...".

Reply:
I think we should be precise in the enum definitions as to what we mean by "above the barrier". Is it > or >=?

Reply:
Good point, I have changed this now.